aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2286,18 +2286,36 @@ impl AptosVM {
         txn: &SignedTransaction,
         log_context: &AdapterLogSchema,
     ) -> (VMStatus, VMOutput) {
-        match self.execute_user_transaction_with_custom_gas_meter(
-            resolver,
-            code_storage,
-            txn,
-            log_context,
-            make_prod_gas_meter,
-        ) {
-            Ok((vm_status, vm_output, _gas_meter)) => (vm_status, vm_output),
-            Err(vm_status) => {
-                let vm_output = discarded_output(vm_status.status_code());
-                (vm_status, vm_output)
-            },
+        if cfg!(feature = "benchmark") && !is_gas_metering_enabled() {
```

Comment:
I'm generally a bit concerned whether we could have some misconfiguration and causes this feature to go public. I would suggest at least in the node initialization logic check that the gas is indeed enabled with benchmark flag set to false in this case?

Reply:
+1, hence I used cargo feature here. Good point about the node config, do you have a code pointer?

I think we can also check if `!cfg!(feature = "benchmark")` is always metered for extra safety.

Reply:
https://github.com/aptos-labs/aptos-core/blob/e72524a56a2fa881ca0b8e2f9c3420223a39851c/aptos-node/src/main.rs#L16

I believe victor also had a similar check here.